.github/workflows/ci-release.yml

```diff
@@ -0,0 +1,31 @@
+name: ci-release
+
+# Run this workflow on push events (i.e. PR merge) to main or release branches,
+# push events for new semantic version tags, and all PRs.
+on:
+  push:
+    branches:
+      - main
+      - "v[0-9]+.x"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+  pull_request:
+
+jobs:
+  lint:
+    uses: ./.github/workflows/lint.yml
+
+  markdown-linter:
+    uses: ./.github/workflows/markdown-linter.yml
+
+  test:
+    uses: ./.github/workflows/test.yml
+
+  goreleaser:
+    uses: ./.github/workflows/goreleaser.yml
+    permissions: write-all
+    secrets:
+      GORELEASER_ACCESS_TOKEN: ${{ secrets.GORELEASER_ACCESS_TOKEN }}
```

Comment:
Since this repo will only contain a Go library, I don't think we need GoReleaser to create pre-built binaries.

```suggestion
```